### PR TITLE
Add check for negative dimension in view

### PR DESCRIPTION
**Pull request by ignaciosica**
fixes: #5774

Is it okay to check at a view level?

~~Regression tests pending~~

### CODE REVIEW

**Comment by chenyuxyz** [comment]
i think it should check in Tensor level, if the arg to a function is invalid, we want to check before passing the arg to tinygrad internal

**Comment by ignaciosica** [comment]
Sure, moved check to Tensor level and leaved an assert in view create

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -441,6 +443,7 @@ def full(shape:Tuple[sint, ...], fill_value:ConstType, **kwargs):
     print(Tensor.full((2, 3), False).numpy())
     ```
     """
+    if not all(s >= 0 for s in argfix(shape)): raise RuntimeError(f"Trying to create view with negative dimension: {shape=}")
```
Comment (chenyuxyz) [inline tinygrad/tensor.py:446]: argfix is called twice, also don't just copy paste error messages. the message does not make sense in Tensor

**Comment by geohot** [comment]
This shouldn't be in there three times. I'm fine with having the assert internally only, and I like the tests.

File: tinygrad/tensor.py (hunk 1 above)
Comment (ignaciosica) [reply tinygrad/tensor.py]: Also, python won't let me use walrus operator in that context:

```
if not all(s >= 0 for s in (new_shape := argfix(shape))): raise RuntimeError(f"Trying to create Tensor with negative dimension: {shape=}")
```
highlights `:=` as having wrong

**Comment by ignaciosica** [comment]
> This shouldn't be in there three times. I'm fine with having the assert internally only, and I like the tests. They pass with only the internal assert?

If the assert is changed for RuntimeError they do, or by changing tests accordingly

**Comment by geohot** [comment]
`RuntimeError` is preferred if it can happen if the user does something wrong, so in this case.

**Pull request merged**